File: Solutions in Javascript/PlusOne.js
/**
 * @author : Anmol Agrawal
 */
/**
 * @param {number[]} digits
 * @return {number[]}
 */
var plusOne = function(digits) {
    var carry = 1;
    for(var i = digits.length - 1; i >= 0; i--){
        var a = digits[i] + carry;
        digits[i] = a % 10;
        carry = a / 10;
        carry = Math.floor(carry);
        if(carry === 0){
            break;
        }
    }
    if(carry > 0){
        digits.unshift(carry);
    }
    return digits;
};

File: Solutions in Java/PlusOne.java
/**
 * @author : Anmol Agrawal
 */
class Solution {
    public int[] plusOne(int[] digits) {
        int carry = 1;
        for(int i = digits.length - 1; i >= 0; i--){
            int a = digits[i] + carry;
            digits[i] = a % 10;
            carry = a / 10;
            carry = (int)Math.floor(carry);
            if(carry == 0){
                break;
            }
        }
        if(carry > 0){
            int[] temp = new int[digits.length + 1];
            temp[0] = carry;
            return temp;
        }
        return digits;
    }
}